x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/user_limit_callout/index.tsx

```diff
@@ -0,0 +1,159 @@
+          data-test-subj="manage-data-sources-button-exceeded"
+        >
+          <FormattedMessage
+            id="xpack.securitySolution.entityAnalytics.privilegedUserMonitoring.userLimit.manageDataSources"
+            defaultMessage="Manage data sources"
+          />
+        </EuiButton>
+      </EuiCallOut>
+    );
+  }
+
+  if (isNearLimit && variant === 'full') {
+    return (
+      <EuiCallOut
+        announceOnMount
+        title={
+          <FormattedMessage
+            id="xpack.securitySolution.entityAnalytics.privilegedUserMonitoring.userLimit.nearLimit.title"
+            defaultMessage="Approaching user limit"
+          />
+        }
+        color="warning"
+        iconType="alert"
+        data-test-subj="privileged-user-near-limit-callout"
+      >
+        <EuiText size="s">
+          <FormattedMessage
+            id="xpack.securitySolution.entityAnalytics.privilegedUserMonitoring.userLimit.nearLimit.description"
+            defaultMessage="You're currently monitoring {currentCount} out of {maxAllowed} allowed privileged users ({utilizationPercentage}% of limit). Consider removing some users to prevent performance degradation."
+            values={{
+              currentCount: <strong>{currentCount.toLocaleString()}</strong>,
+              maxAllowed: <strong>{maxAllowed.toLocaleString()}</strong>,
+              utilizationPercentage: <strong>{utilizationPercentage}</strong>,
+            }}
+          />
+        </EuiText>
+        <EuiSpacer size="s" />
+        <EuiSpacer size="s" />
+        <EuiButton
+          size="s"
+          iconType="gear"
+          color="warning"
+          onClick={onManageDataSources || (() => {})}
+          data-test-subj="manage-data-sources-button-near-limit"
+        >
+          <FormattedMessage
+            id="xpack.securitySolution.entityAnalytics.privilegedUserMonitoring.userLimit.manageDataSources"
+            defaultMessage="Manage data sources"
+          />
+        </EuiButton>
+      </EuiCallOut>
+    );
+  }
+
+  if (variant === 'compact') {
+    return (
+      <EuiCallOut
+        announceOnMount
+        title={
+          <FormattedMessage
+            id="xpack.securitySolution.entityAnalytics.privilegedUserMonitoring.userLimit.info.title"
+            defaultMessage="User limit information"
+          />
+        }
+        color="primary"
+        iconType="info"
+        size="s"
+        data-test-subj="privileged-user-limit-info-callout"
+      >
+        <EuiText size="s">
+          <FormattedMessage
+            id="xpack.securitySolution.entityAnalytics.privilegedUserMonitoring.userLimit.info.description"
+            defaultMessage="Maximum supported number of privileged users allowed: {maxAllowed}"
+            values={{
+              maxAllowed: <strong>{maxAllowed.toLocaleString()}</strong>,
+            }}
+          />
+        </EuiText>
+      </EuiCallOut>
+    );
+  }
+
+  return null;
```

Comment:
nit: Moving the return null to the beginning of the function usually makes it easier to read.